common/src/main/java/com/wynntils/core/features/UserFeature.java

```diff
@@ -4,12 +4,24 @@
  */
 package com.wynntils.core.features;
 
+import com.wynntils.core.config.Configurable;
+import com.wynntils.core.config.objects.ConfigOptionHolder;
 import com.wynntils.core.config.properties.ConfigOption;
 
 /**
  * A feature that is enabled & disabled by the user.
  */
-public abstract class UserFeature extends Feature {
+public abstract class UserFeature extends Feature implements Configurable {
     @ConfigOption(displayName = "Enabled", description = "Should this feature be enabled?")
     protected boolean userEnabled = true;
+
+    /** This handles the user enabling/disabling a feature in-game */
+    @Override
+    public void onConfigUpdate(ConfigOptionHolder option) {
```

Comment:
Don't you need to check if the current option is the enabled one?  
Either that or I don't get something about this method

Reply:
This will run every time an update happens, yes, but `tryEnable` and `tryDisable` don't do anything if the feature is already in the correct state. I can add a check for the field's name though

Reply:
Why would we try to enable a feature by changing an unrelated setting? 

Also, if it is not user enabled, why do we try to disable it?

Reply:
Because if it's not user enabled, the feature is supposed to be disabled

Reply:
This doesn't respect the condition system in place

Reply:
Yes, it does - `tryEnable` calls `enable`, which fails if the conditions have not been met already

Reply:
Correct me if I am wrong, but here, what you are trying to achieve is an update listener for the `userEnabled` field?

If so, I would definitely add a check for the field name. Also, what happens if this get's overwritten by the feature? I don't (and we shouldn't) expect the feature to re-add this logic to the method (or call super).

Reply:
> Yes, it does - `tryEnable` calls `enable`, which fails if the conditions have not been met already

Oh looks like I forgot a few modifications to `Feature`, nice.

Reply:
Maybe we could automatically call the super method through reflection, but that seems scuffed.

Reply:
I went ahead and added a name check. My plan for dealing with subclasses was just super calls, yeah - I don't think there's a good way around that unless we make an additional wrapper method in `UserFeature` that gets called from `onConfigUpdate`

Reply:
I feel like the super calls will absolutely be forgotten when actually writing code, although I can't think of something better atm.

Reply:
The javax library contains the annotation `@OverridingMethodMustInvokeSuper`, which raises an error if the super method isn't called. We should be able to use this for peace of mind

Reply:
Hm, looks like the annotation isn't fully implemented - although it will still cause a warning in intellij

Reply:
Warning should be enough

Reply:
Once again, I'm not convinced the "turn feature on or off" thing really should be a config like the others. It could just as well be seen as configurations on the FeatureRegistry. It has a bit of an odd ring to it, that a feature is configured to turn itself on or off. I can very much imagine a GUI where the "on/off" functionality of a feature is shown by a switch or checkbox next to the feature name in a list, but the configuration variables for that feature is shown on a config pane to the right of the list. Like the common model for how you enable/disable mods/plugins etc in one place, but configure them elsewhere.

Especially if this switch is making the "real" configuration options more complicated, by forcing the use of a super() call. I think this is a sign that feature-enabled is a property that should have special treatment, and not be considered a general config option.